src/frontend/src/optimizer/plan_node/stream_dynamic_filter.rs

```diff
@@ -100,7 +98,13 @@ impl StreamDynamicFilter {
                     // downstream. See `derive_watermark_columns`.
                     true
                 }
-                _ => false,
+                ExprType::LessThan | ExprType::LessThanOrEqual => {
+                    // For <= and <, watermark on rhs means all rows older than the watermark should already be emitted,
+                    // and future lhs inputs should be directly passed to downstream without any state table operation.
+                    // So, the state table can be cleaned up.
+                    true
+                }
```

Comment:
Now the left table is cleaned by rhs watermark when comparator is `<` and `<=` as well.